src/com/uwsoft/editor/gdx/ui/dialogs/CustomVariablesDialog.java

```diff
@@ -20,179 +20,185 @@
-                }
-            });
-
-
-            itm.getTextButtonById("deleteBtn").addListener(new ClickListener() {
-                public boolean touchDown (InputEvent event, float x, float y, int pointer, int button) {
-                    return true;
-                }
-                public void touchUp (InputEvent event, float x, float y, int pointer, int button) {
-                    vars.removeVariable(key);
-                    renderMainList();
-                }
-            });
-        }
-    }
+public class CustomVariablesDialog extends VisWindow {
+	
+	private IBaseItem item;
+	private CustomVariables vars;
+	private VisList<String> varNames;
+	
+	private VisTextField varName;
+	private VisTextField varVal;
+	
+	private VisTextButton changeName;
+	private VisTextButton updateValue;
+	private VisTextButton deleteVar;
+	
+	private VisTextField keyName;
+	private VisTextField keyVal;
+	private VisTextButton addVar;
+	
+	public CustomVariablesDialog(final IBaseItem item) {
+		super("Custom Variables Dialog");
+		
+		addCloseButton();
+		closeOnEscape();
+		
+		TableUtils.setSpacingDefaults(this);
+		columnDefaults(0).left();
+		this.item = item;
+		vars = item.getCustomVariables();
+		
+		buildLayout();
+		
+		setupHooks();
+		
+		pack();
+		centerWindow();
+	}
+	
+	private void buildLayout() {
+		// Left Side list of Variables
+		varNames = new VisList<String>();
+		renderNames();
+		VisScrollPane pane = new VisScrollPane(varNames);
+		pane.setFlickScroll(false);
+		pane.setFadeScrollBars(false);
+		varNames.setSelectedIndex(-1);
+		
+		// Used Labels in layout.
+		VisLabel lbl1 = new VisLabel("Variable Name:");
+		VisLabel lbl2 = new VisLabel("Variable Value:");
+		VisLabel lbl3 = new VisLabel("New Variable:");
+		VisLabel lbl4 = new VisLabel("=");
+		
+		// Text Fields in layout.
+		varName = new VisTextField("");
+		varName.setDisabled(true);
+		varVal = new VisTextField("");
+		keyName = new VisTextField("");
+		keyVal = new VisTextField("");
+		
+		// Buttons in layout.
+		changeName = new VisTextButton("Change Name");
+		updateValue = new VisTextButton("Update Value");
+		deleteVar = new VisTextButton("Delete Variable");
+		addVar = new VisTextButton("Add Variable");
+		
+		// Layout everything
+		VisTable form = new VisTable(true);
+		form.add(lbl1);
+		form.add(varName).expandX().fillX();
+		form.add(changeName);
+		form.row();
+		
+		form.add(lbl2);
+		form.add(varVal).expandX().fillX();
+		form.row();
+		
+		form.add(updateValue);
+		form.add(deleteVar);
+		
+		VisTable newKVE = new VisTable(true);
+		newKVE.add(lbl3);
+		newKVE.add(keyName);
+		newKVE.add(lbl4);
+		newKVE.add(keyVal);
+		newKVE.add(addVar);
+		
+		add(pane).expandX().expandY().fillX().fillY();
+		add(new Separator()).padTop(10).fillY().expandY();
+		add(form).row();
+		
+		add(newKVE).colspan(3).expandX().fillX().padBottom(3).padTop(3).row();
+	}
+	
+	private void setupHooks() {
+		varNames.addListener(new ChangeListener() {
```

Comment:
After changing the variable, you need to "update" the item, by calling item.updateDataVO();